src/main/scala/io/iohk/ethereum/transactions/PendingTransactionsManager.scala

```diff
@@ -44,11 +50,12 @@ class PendingTransactionsManager(peerManager: ActorRef, peerMessageBus: ActorRef
   peerMessageBus ! Subscribe(MessageClassifier(Set(SignedTransactions.code), PeerSelector.AllPeers))
 
   override def receive: Receive = {
-    case AddTransaction(signedTransaction) =>
-      if (!pendingTransactions.contains(signedTransaction)) {
-        pendingTransactions :+= signedTransaction
+    case AddTransactions(signedTransactions) =>
+      val transactionsToAdd = signedTransactions.filterNot(t => pendingTransactions.contains(t))
+      if (transactionsToAdd.nonEmpty) {
+        pendingTransactions = (pendingTransactions ++ transactionsToAdd).takeRight(miningConfig.txPoolSize)
```

Comment:
should we implement a strategy for which transactions do we drop? like: dropHead or dropTail ?